packages/core/src/render/api.ts

```diff
@@ -286,6 +290,26 @@ export abstract class Renderer2 {
 }
 
 
-export const SWITCH_RENDERER2_FACTORY__POST_R3__ = render3InjectRenderer2;
+export const SWITCH_RENDERER2_FACTORY__POST_R3__ = injectRenderer2;
 const SWITCH_RENDERER2_FACTORY__PRE_R3__ = noop;
-const SWITCH_RENDERER2_FACTORY: typeof render3InjectRenderer2 = SWITCH_RENDERER2_FACTORY__PRE_R3__;
+const SWITCH_RENDERER2_FACTORY: typeof injectRenderer2 = SWITCH_RENDERER2_FACTORY__PRE_R3__;
+
+/** Returns a Renderer2 (or throws when application was bootstrapped with Renderer3) */
+function getOrCreateRenderer2(view: LView): Renderer2 {
+  const renderer = view[RENDERER];
+  if (isProceduralRenderer(renderer)) {
+    return renderer as Renderer2;
+  } else {
+    throw new Error('Cannot inject Renderer2 when the application uses Renderer3!');
+  }
```

Comment:
We can probably use `ngDevMode` here:
```suggestion
  } else if (ngDevMode) {
    throw new Error('Cannot inject Renderer2 when the application uses Renderer3!');
  }
```

Reply:
fixed